controller/instance_manager_controller.go

```diff
@@ -2044,6 +2087,265 @@ func (m *InstanceManagerMonitor) StopMonitorWithLock() {
 	m.done = true
 }
 
+func (m *InstanceManagerMonitor) syncOrphans(im *longhorn.InstanceManager, instanceProcess map[string]longhorn.InstanceProcess) {
+	engineProcess := map[string]longhorn.InstanceProcess{}
+	replicaProcess := map[string]longhorn.InstanceProcess{}
+	for name, process := range instanceProcess {
+		switch process.Status.Type {
+		case longhorn.InstanceTypeEngine:
+			engineProcess[name] = process
+		case longhorn.InstanceTypeReplica:
+			replicaProcess[name] = process
+		}
+	}
+
+	existOrphans, err := m.getNodeRuntimeOrphanMap()
+	if err != nil {
+		m.logger.WithError(err).Errorf("Failed to list orphans on node %s", m.controllerID)
+		return
+	}
+
+	// create orphan CR for orphaned instances
+	for _, engineInstance := range engineProcess {
+		if isOrphaned, err := m.isEngineOrphaned(engineInstance.Spec.Name); err != nil {
+			m.logger.WithError(err).Errorf("Failed to check orphan for engine instance %v", engineInstance.Spec.Name)
+		} else if isOrphaned {
+			m.logger.Infof("Engine instance %v is orphaned", engineInstance.Spec.Name)
+			err := m.createOrphan(im, engineInstance.Spec.Name, longhorn.OrphanTypeEngineInstance, engineInstance.Spec.DataEngine)
+			if err != nil {
+				m.logger.WithError(err).Errorf("Failed to create orphan for engine instance %v", engineInstance.Spec.Name)
+			}
+		}
+	}
+	for _, replicaInstance := range replicaProcess {
+		if isOrphaned, err := m.isReplicaOrphaned(replicaInstance.Spec.Name); err != nil {
+			m.logger.WithError(err).Errorf("Failed to check orphan for replica instance %v", replicaInstance.Spec.Name)
+		} else if isOrphaned {
+			m.logger.Infof("Replica instance %v is orphaned", replicaInstance.Spec.Name)
+			err := m.createOrphan(im, replicaInstance.Spec.Name, longhorn.OrphanTypeReplicaInstance, replicaInstance.Spec.DataEngine)
+			if err != nil {
+				m.logger.WithError(err).Errorf("Failed to create orphan for replica instance %v", replicaInstance.Spec.Name)
+			}
+		}
+	}
+
+	// update instance state on orphan CRs
```

Comment:
Looks like this commend is no longer needed since we don't update the orphan status inside IM controller now?